### PR TITLE
issue-1472:Add statement to check if mCacheStats isInitialized when getSize() or getCount()

### PR DESCRIPTION
**Pull request by eldk**
Hello,

#1472

Add statement to check if mCacheStats isInitialized in getSize() and getCount(), if not, call maybeUpdateFileCacheSize().

Thanks,

Eric

### CODE REVIEW

**Comment by lambdapioneer** [comment]
Hi @eldk,

thanks for the PR. Looks good 👍  Let me import it to run the tests as CI is broken at the current base of your diff.

**Comment by facebook-github-bot-0** [comment]
Thanks for importing. If you are a Facebook employee, you can view this diff [on Phabricator](https://phabricator.intern.facebook.com/D3913333).

**Comment by facebook-github-bot-4** [comment]
@eldk updated the pull request - [view changes](https://github.com/facebook/fresco/pull/1475/files/ad0732d073e65e404ae376448681b8d2102143b1..eb4451f9a97346e120df89aae6604efb4eb20e4e) - [changes since last import](https://github.com/facebook/fresco/pull/1475/files/ad0732d073e65e404ae376448681b8d2102143b1..eb4451f9a97346e120df89aae6604efb4eb20e4e)

**Comment by eldk** [comment]
https://github.com/facebook/fresco/blob/master/imagepipeline-base/src/test/java/com/facebook/cache/disk/DiskStorageCacheTest.java#L298

File: imagepipeline-base/src/main/java/com/facebook/cache/disk/DiskStorageCache.java [hunk 1]
```diff
@@ -592,10 +592,16 @@ private void updateFileCacheSizeLimit() {
   }
 
   public long getSize() {
+    if (!mCacheStats.isInitialized()){
+      maybeUpdateFileCacheSize();
```
Comment (lambdapioneer) [inline imagepipeline-base/src/main/java/com/facebook/cache/disk/DiskStorageCache.java]: We would have to have a `synchronized(mLock)` around this, as `maybeUpdateFileCacheSize()` is expected to be called in a synchronized manner (see line 703/709).

Alternatively, we could throw an `IllegalStateException` in those cases where the client has to wait for the initialization.

Which variant would you prefer? And what the use-cases would you anticipate where clients need to check this early on?

File: imagepipeline-base/src/main/java/com/facebook/cache/disk/DiskStorageCache.java [hunk 2]
```diff
@@ -592,10 +592,16 @@ private void updateFileCacheSizeLimit() {
...
+    }
     return mCacheStats.getSize();
   }
 
   public long getCount() {
+    if (!mCacheStats.isInitialized()){
+      maybeUpdateFileCacheSize();
```
Comment (lambdapioneer) [inline imagepipeline-base/src/main/java/com/facebook/cache/disk/DiskStorageCache.java]: same here

**Comment by eldk** [comment]
https://github.com/facebook/fresco/issues/1460

**Comment by eldk** [comment]
Hello @lambdapioneer  ,

Setting DiskCacheConfig as this : 

`DiskCacheConfig diskCacheConfig = DiskCacheConfig.newBuilder(context)`
(...)
`.setIndexPopulateAtStartupEnabled(true)`
`.build();`

Doc : http://frescolib.org/javadoc/reference/com/facebook/cache/disk/DiskCacheConfig.html

Solve this : 
https://github.com/facebook/fresco/issues/1472 (getSize and getCount return -1 when app is restarted).

So this PR is not needed.

Sorry,

Eric

**Pull request closed without merging**